dotnet/samples/KernelSyntaxExamples/Example09_FunctionTypes.cs

```diff
@@ -95,109 +82,45 @@ public async Task<string> Type03Async()
-    public async Task<ContextVariables> Type07Async(ContextVariables variables)
-    {
-        await Task.Delay(0);
-        Console.WriteLine("Running function type 7");
-        return variables;
-    }
-
-    [KernelFunction]
-    public void Type08(string x)
-    {
-        Console.WriteLine("Running function type 8");
-    }
-
-    [KernelFunction]
-    public string Type09(string x)
-    {
-        Console.WriteLine("Running function type 9");
-        return "";
-    }
-
-    [KernelFunction]
-    public async Task<string> Type10Async(string x)
+    public async Task<string> Type04Async(Kernel kernel)
     {
-        await Task.Delay(0);
-        Console.WriteLine("Running function type 10");
+        var summary = await kernel.InvokeAsync(kernel.Plugins["SummarizePlugin"]["Summarize"], new KernelFunctionArguments() { { "input", "blah blah blah" } });
+        Console.WriteLine($"Running function type 4 [{summary?.GetValue<string>()}]");
         return "";
     }
 
     [KernelFunction]
-    public void Type11(string x, ContextVariables variables)
+    public void Type05(string x)
     {
-        Console.WriteLine("Running function type 11");
+        Console.WriteLine("Running function type 5");
     }
 
     [KernelFunction]
-    public string Type12(string x, ContextVariables variables)
+    public string Type06(string x)
     {
-        Console.WriteLine("Running function type 12");
+        Console.WriteLine("Running function type 6");
         return "";
     }
 
     [KernelFunction]
-    public async Task<string> Type13Async(string x, ContextVariables variables)
+    public async Task<string> Type07Async(string x)
     {
         await Task.Delay(0);
-        Console.WriteLine("Running function type 13");
+        Console.WriteLine("Running function type 07");
         return "";
     }
 
     [KernelFunction]
-    public async Task<ContextVariables> Type14Async(string x, ContextVariables variables)
-    {
-        await Task.Delay(0);
-        Console.WriteLine("Running function type 14");
-        return variables;
-    }
-
-    [KernelFunction]
-    public async Task Type15Async(string x)
-    {
-        await Task.Delay(0);
-        Console.WriteLine("Running function type 15");
-    }
-
-    [KernelFunction]
-    public async Task Type16Async(ContextVariables variables)
-    {
-        await Task.Delay(0);
-        Console.WriteLine("Running function type 16");
-    }
-
-    [KernelFunction]
-    public async Task Type17Async(string x, ContextVariables variables)
+    public async Task Type08Async(string x)
     {
         await Task.Delay(0);
-        Console.WriteLine("Running function type 17");
+        Console.WriteLine("Running function type 08");
     }
 
     [KernelFunction]
-    public async Task Type18Async()
+    public async Task Type09Async()
     {
         await Task.Delay(0);
-        Console.WriteLine("Running function type 18");
+        Console.WriteLine("Running function type 09");
     }
```

Comment:
Should there also be

```csharp
public async FunctionResult Type10()
public async Task<FunctionResult> Type11Async()
```

Reply:
They can easily be be added. We will have to extend existing marshaling for function results to prevent function results be wrapped into another function result. I'm happy to update it.